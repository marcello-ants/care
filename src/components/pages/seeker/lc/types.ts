import { GeneralProviderProfile } from '@/types/seeker';

export interface ProviderProfile extends GeneralProviderProfile {
  numberOfReviews: number;
  qualities?: ProviderQualities;
  services?: ProviderServices;
  firstName: string;
}

export interface ProviderQualities {
  certifiedNursingAssistant?: boolean;
  certifiedRegisterNurse?: boolean;
  certifiedTeacher?: boolean;
  childDevelopmentAssociate?: boolean;
  comfortableWithPets?: boolean;
  cprTrained?: boolean;
  crn?: boolean;
  doesNotSmoke?: boolean;
  doula?: boolean;
  earlyChildDevelopmentCoursework?: boolean;
  earlyChildhoodEducation?: boolean;
  firstAidTraining?: boolean;
  nafccCertified?: boolean;
  ownTransportation?: boolean;
  specialNeedsCare?: boolean;
  trustlineCertifiedCalifornia?: boolean;
}

export interface ProviderServices {
  carpooling?: boolean;
  craftAssistance?: boolean;
  errands?: boolean;
  groceryShopping?: boolean;
  laundryAssistance?: boolean;
  lightHousekeeping?: boolean;
  mealPreparation?: boolean;
  swimmingSupervision?: boolean;
  travel?: boolean;
}
