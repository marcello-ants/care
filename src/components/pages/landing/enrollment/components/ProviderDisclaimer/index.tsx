// External Dependencies
import { Typography, Link } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

export default function ProviderDisclaimer() {
  // Styles
  const classes = useStyles();

  return (
    <Typography careVariant="disclaimer1" className={classes.disclaimer}>
      <span>
        You must be 18 years or older to use Care.com. By clicking &quot;Submit,&quot; you agree to
        our
      </span>{' '}
      <Link href="/terms-of-use-p1012.html" careVariant="disclaimer1">
        Terms of Use
      </Link>{' '}
      <span>and</span>{' '}
      <Link href="/privacy-policy-p1013.html" careVariant="disclaimer1">
        Privacy Policy
      </Link>
      <span>.</span>
    </Typography>
  );
}
