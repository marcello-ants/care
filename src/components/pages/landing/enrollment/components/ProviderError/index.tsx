// External Dependencies
import { Box } from '@material-ui/core';
import { Icon24UtilityWarning } from '@care/react-icons';
import { Typography } from '@care/react-component-lib';

// Styles
import useStyles from './styles';

function ProviderError() {
  // Styles
  const classes = useStyles();

  return (
    <Box className={classes.errorBanner}>
      <Box>
        <Icon24UtilityWarning color="#CF230C" />
      </Box>
      <Box>
        <Typography variant="h4">Whoops! Something went wrong</Typography>
        <Typography careVariant="body3">
          There was an error when we tried to submit your request. Please try again.
        </Typography>
      </Box>
    </Box>
  );
}

export default ProviderError;
