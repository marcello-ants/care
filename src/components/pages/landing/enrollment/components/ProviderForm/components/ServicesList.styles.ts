import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  empty: {
    '& > .MuiSelect-select': {
      color: theme.palette.care?.grey[700],
      fontWeight: 'normal',
    },
  },
  inputContainer: {
    width: '100%',
    '& > div': {
      padding: 0,
    },
  },
  error: {
    color: theme?.palette?.care?.red[800],
    marginLeft: theme.spacing(1.25),
    marginTop: theme.spacing(0.5),
  },
}));

export default useStyles;
