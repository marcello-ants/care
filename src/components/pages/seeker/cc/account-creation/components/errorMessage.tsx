import React from 'react';
import { Box, Grid } from '@material-ui/core';
import { Alert } from '@material-ui/lab';

type ErrorMessageData = {
  errorMessageText: string;
};

const ErrorMessageNotification = (props: ErrorMessageData) => {
  const { errorMessageText } = props;

  return (
    <Grid item xs={12}>
      <Box mb={3}>
        <Alert severity="error">{errorMessageText}</Alert>
      </Box>
    </Grid>
  );
};

export default ErrorMessageNotification;
