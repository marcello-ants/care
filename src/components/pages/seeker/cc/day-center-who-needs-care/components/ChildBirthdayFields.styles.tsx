// External Dependencies
import { makeStyles } from '@material-ui/core';
import { theme } from '@care/material-ui-theme';

const useStyles = makeStyles({
  childContainer: {
    padding: theme.spacing(0),
  },
  childContainerNth: {
    marginTop: theme.spacing(4),
    position: 'relative',
  },
  optionsContainer: {
    display: 'flex',
    flexDirection: 'row',
  },
  fullWidth: {
    width: '100%',
  },
  error: {
    color: theme.palette.care.red[800],
    paddingLeft: theme.spacing(1),
  },
  selectContainer: {
    marginTop: theme.spacing(1),
  },
  monthSelect: {
    marginRight: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  yearSelect: {
    marginLeft: theme.spacing(0.5),
    '& label': {
      paddingLeft: theme.spacing(1),
    },
  },
  removeChildButton: {
    position: 'absolute',
    top: '0',
    right: '0',
    width: 'auto',
    height: 'auto',
    lineHeight: 'unset',
    minWidth: 'auto',
    padding: theme.spacing(0),
    color: theme.palette.care.blue[700],
    '&:hover': {
      backgroundColor: 'transparent',
    },
  },
});

export default useStyles;
