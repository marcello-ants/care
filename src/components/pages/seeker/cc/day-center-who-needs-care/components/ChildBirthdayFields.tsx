// External Dependencies
import React, { useMemo } from 'react';
import { Button, FormHelperText, Grid, InputLabel, MenuItem } from '@material-ui/core';
import { Select, Typography } from '@care/react-component-lib';

// Internal Dependencies
import { ChildDateOfBirth } from '@/types/seekerCC';
import { getBirthAvailableMonthsAndYears } from '../utils';
import useStyles from './ChildBirthdayFields.styles';

// Interfaces
interface TErrorProps {
  year?: string;
  month?: string;
}

type SelectInputProps = React.ComponentProps<typeof Select>;
interface IChildBirthdayFieldsProps {
  child: ChildDateOfBirth;
  onRemoveChild: () => void;
  onChange: SelectInputProps['onChange'];
  childIndex: number;
  error?: TErrorProps;
  allow9MonthsOutFromNow?: boolean;
}

function ChildBirthdayFields({
  child,
  childIndex: i,
  onChange,
  onRemoveChild,
  error,
  allow9MonthsOutFromNow,
}: IChildBirthdayFieldsProps) {
  const classes = useStyles();

  const { months, years } = useMemo(
    () => getBirthAvailableMonthsAndYears(child.month, child.year, allow9MonthsOutFromNow),
    [child.year, child.month, allow9MonthsOutFromNow]
  );

  const monthsOptions = useMemo(() => {
    return months.map((month) => {
      return (
        <MenuItem key={month.value} value={month.value}>
          {month.name}
        </MenuItem>
      );
    });
  }, [months]);

  const yearsOptions = useMemo(() => {
    return years.map((year: string) => {
      return (
        <MenuItem key={year} value={year}>
          {year}
        </MenuItem>
      );
    });
  }, [years]);

  return (
    <Grid
      // eslint-disable-next-line react/no-array-index-key
      item
      xs={12}
      className={`${
        i === 0
          ? `${classes.childContainer}`
          : `${classes.childContainer} ${classes.childContainerNth}`
      }`}>
      <Typography variant="h3">Child {i + 1}</Typography>
      {i !== 0 && (
        <Button className={classes.removeChildButton} onClick={onRemoveChild}>
          Remove Child
        </Button>
      )}
      <Grid item xs={12} className={classes.optionsContainer}>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.monthSelect}`}>
          <InputLabel id={`month-select-label-${i}`}>Birth month</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`month-select-label-${i}`}
            id={`month-select-child-${i}`}
            name={`children[${i}].month`}
            onChange={onChange}
            value={child.month}
            error={Boolean(error?.month)}>
            {monthsOptions}
          </Select>
          {error?.month && (
            <FormHelperText className={classes.error}>Select their birth month</FormHelperText>
          )}
        </Grid>
        <Grid item xs={6} className={`${classes.selectContainer} ${classes.yearSelect}`}>
          <InputLabel id={`year-select-label-${i}`}>Birth year</InputLabel>
          <Select
            className={classes.fullWidth}
            labelId={`year-select-label-${i}`}
            id={`year-select-child-${i}`}
            name={`children[${i}].year`}
            onChange={onChange}
            value={child.year}
            error={Boolean(error?.year)}>
            {yearsOptions}
          </Select>
          {error?.year && (
            <FormHelperText className={classes.error}>Select their birth year</FormHelperText>
          )}
        </Grid>
      </Grid>
    </Grid>
  );
}

ChildBirthdayFields.defaultProps = {
  error: undefined,
  allow9MonthsOutFromNow: undefined,
};

export type { IChildBirthdayFieldsProps, TErrorProps };
export default ChildBirthdayFields;
