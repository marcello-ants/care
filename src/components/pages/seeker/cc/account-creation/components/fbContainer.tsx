import { Box, makeStyles, Grid, Button } from '@material-ui/core';
import { useFacebookApiKey } from '@/components/FacebookApiKeyContext';

// @ts-ignore
import FacebookLogin from 'react-facebook-login/dist/facebook-login-render-props';
import { Icon48SocialMediaFacebook } from '@care/react-icons';

export type facebookResponseProps = {
  accessToken: string;
  // eslint-disable-next-line
  data_access_expiration_time?: number;
  email: string;
  expiresIn?: number;
  graphDomain?: string;
  id?: string;
  name: string;
  signedRequest?: string;
  userID?: string;
};

export interface SplitFormValues {
  email: string;
}

export interface SplitFormWithFbData extends SplitFormValues {
  firstName?: string;
  lastName?: string;
  accessToken?: string;
}

const mobileWidth = 360;
const fbColor = '#1877F2';
const fbIconSize = '35px';
const useStyles = (fbEmphasizedDesign: boolean) =>
  makeStyles((theme) => ({
    fbButton: {
      paddingTop: theme.spacing(5),
      borderColor: fbEmphasizedDesign ? fbColor : theme.palette.default?.dark,
      background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      color: fbEmphasizedDesign ? theme.palette.care?.white : theme.palette.default?.dark,
      fontSize: fbEmphasizedDesign ? '21px' : '18px',
      '&:hover': {
        background: fbEmphasizedDesign ? fbColor : theme.palette.care?.white,
      },
    },
    fbButtonContent: {
      display: 'grid',
      gridTemplateColumns: 'repeat(3, minmax(0, 1fr))',
      [theme.breakpoints.down(mobileWidth)]: {
        display: 'flex',
        alignItems: 'center',
      },
    },
    fbIconContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: fbIconSize,
      '& svg': {
        width: fbIconSize,
        height: fbIconSize,
      },
    },
    fbIcon: {
      background: '#1a77f2',
      borderRadius: '19px',
    },
    fbButtonText: {
      justifySelf: 'center',
      marginLeft: '10px',
      marginRight: '5px',
      fontColor: theme.palette.care?.white,
    },
  }));

type GetFbContainerProps = {
  handleFacebookLogin: (response: facebookResponseProps) => Promise<void>;
  fbButtonOnTop: boolean;
  fbButtonOnBottom: boolean;
};

const getFbContainer = ({
  handleFacebookLogin,
  fbButtonOnTop,
  fbButtonOnBottom,
}: GetFbContainerProps) => {
  const fbEmphasizedDesign = fbButtonOnBottom || fbButtonOnTop;
  const classes = useStyles(fbEmphasizedDesign)();
  const { apiKey } = useFacebookApiKey();

  return (
    <Grid item xs={12}>
      <Box pt={fbButtonOnTop ? 0 : 2}>
        <FacebookLogin
          appId={apiKey}
          autoLoad={false}
          fields="name,email"
          callback={handleFacebookLogin}
          render={(renderProps: any) => (
            <Button
              onClick={renderProps.onClick}
              className={classes.fbButton}
              variant="outlined"
              size="large"
              fullWidth>
              <div className={classes.fbButtonContent}>
                <div className={classes.fbIconContainer}>
                  <Icon48SocialMediaFacebook size="38px" color="white" className={classes.fbIcon} />
                </div>
                <p className={classes.fbButtonText}>Continue with Facebook</p>
              </div>
            </Button>
          )}
        />
      </Box>
    </Grid>
  );
};

export default getFbContainer;
