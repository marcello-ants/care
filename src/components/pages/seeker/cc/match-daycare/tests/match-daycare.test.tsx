import React from 'react';
import { render } from '@testing-library/react';
import { NextRouter, useRouter } from 'next/router';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { cloneDeep } from 'lodash-es';

// Internal Dependencies
import { AppStateProvider } from '@/components/AppState';
import { FeatureFlagsProvider } from '@/components/FeatureFlagsContext';
import { initialAppState } from '@/state';
import { AppState } from '@/types/app';
import AdditionalSupport from '../match-daycare';

// Data
const TEST_DATA_SUCCESS_ZIP_CODE = '91911';

// Mocks
jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

const flags = {
  'enrollment-mfe-seeker-daycare-recommendation': {
    reason: { kind: 'FALLTHROUGH' },
    value: false,
    variationIndex: 1,
  },
};

let mockRouter: Pick<NextRouter, 'push' | 'asPath' | 'pathname'>;
const clonedAppState = cloneDeep(initialAppState);
const appState: AppState = {
  ...clonedAppState,
  seeker: {
    ...clonedAppState.seeker,
    zipcode: TEST_DATA_SUCCESS_ZIP_CODE,
  },
};

// Utilities Functions

function renderPage() {
  const pathname = '/seeker/cc/match-daycare';
  mockRouter = {
    push: jest.fn(),
    asPath: pathname,
    pathname,
  };
  (useRouter as jest.Mock).mockReturnValue(mockRouter);
  const mocks: MockedResponse[] = [];

  return render(
    <MockedProvider mocks={mocks} addTypename>
      <FeatureFlagsProvider flags={flags}>
        <AppStateProvider initialStateOverride={appState}>
          <AdditionalSupport />
        </AppStateProvider>
      </FeatureFlagsProvider>
    </MockedProvider>
  );
}

describe('Daycare - Match daycares', () => {
  it('matches snapshot', async () => {
    const view = renderPage();
    expect(view.asFragment()).toMatchSnapshot();
  });
});
