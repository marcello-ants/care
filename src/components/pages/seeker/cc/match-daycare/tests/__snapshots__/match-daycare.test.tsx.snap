// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Daycare - Match daycares matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-mainContainer-5 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        class="MuiTypography-root makeStyles-header-8 MuiTypography-h2"
      >
        Looking for daycares in , ...
      </h2>
      <ul
        class="MuiList-root makeStyles-list-1 MuiList-padding"
      >
        <li
          class="MuiListItem-root makeStyles-listItem-2 makeStyles-itemTransitionBegin-6 makeStyles-itemTransitionEnd-7 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-itemIcon-3"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-root-12"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2 12a10 10 0 1110 10A10 10 0 012 12zm1.32 0A8.68 8.68 0 1012 3.32 8.7 8.7 0 003.32 12z"
                fill="#00a979"
              />
              <path
                d="M10.35 15.3a.68.68 0 01-.36-.18l-2.43-2.64a.54.54 0 01.09-.62.45.45 0 01.63 0l2.07 2.29 4.78-5a.45.45 0 01.63 0 .41.41 0 010 .62l-5 5.37a.49.49 0 01-.41.16z"
                fill="#00a979"
              />
              <path
                d="M10.35 15.55a.91.91 0 01-.53-.25l-2.48-2.71a.77.77 0 01.13-.9.7.7 0 011 0l1.9 2.11L15 9a.7.7 0 011 0 .68.68 0 01.21.49.67.67 0 01-.21.49l-5 5.37a.77.77 0 01-.65.2zm-2.58-3.21L10.18 15a.54.54 0 00.18.1c.07 0 .1 0 .18-.1l5-5.37a.19.19 0 00.07-.14.18.18 0 00-.06-.13.19.19 0 00-.28 0l-5 5.21L8.1 12a.2.2 0 00-.28 0 .28.28 0 00-.05.34z"
                fill="#00a979"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Locations nearest to you
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-listItem-2 makeStyles-itemTransitionBegin-6 makeStyles-itemTransitionEnd-7 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-itemIcon-3"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-root-13"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2 12a10 10 0 1110 10A10 10 0 012 12zm1.32 0A8.68 8.68 0 1012 3.32 8.7 8.7 0 003.32 12z"
                fill="#00a979"
              />
              <path
                d="M10.35 15.3a.68.68 0 01-.36-.18l-2.43-2.64a.54.54 0 01.09-.62.45.45 0 01.63 0l2.07 2.29 4.78-5a.45.45 0 01.63 0 .41.41 0 010 .62l-5 5.37a.49.49 0 01-.41.16z"
                fill="#00a979"
              />
              <path
                d="M10.35 15.55a.91.91 0 01-.53-.25l-2.48-2.71a.77.77 0 01.13-.9.7.7 0 011 0l1.9 2.11L15 9a.7.7 0 011 0 .68.68 0 01.21.49.67.67 0 01-.21.49l-5 5.37a.77.77 0 01-.65.2zm-2.58-3.21L10.18 15a.54.54 0 00.18.1c.07 0 .1 0 .18-.1l5-5.37a.19.19 0 00.07-.14.18.18 0 00-.06-.13.19.19 0 00-.28 0l-5 5.21L8.1 12a.2.2 0 00-.28 0 .28.28 0 00-.05.34z"
                fill="#00a979"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Matching your family's needs
            </p>
          </div>
        </li>
        <li
          class="MuiListItem-root makeStyles-listItem-2 makeStyles-itemTransitionBegin-6 makeStyles-itemTransitionEnd-7 MuiListItem-gutters"
        >
          <div
            class="MuiListItemIcon-root makeStyles-itemIcon-3"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-root-14"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2 12a10 10 0 1110 10A10 10 0 012 12zm1.32 0A8.68 8.68 0 1012 3.32 8.7 8.7 0 003.32 12z"
                fill="#00a979"
              />
              <path
                d="M10.35 15.3a.68.68 0 01-.36-.18l-2.43-2.64a.54.54 0 01.09-.62.45.45 0 01.63 0l2.07 2.29 4.78-5a.45.45 0 01.63 0 .41.41 0 010 .62l-5 5.37a.49.49 0 01-.41.16z"
                fill="#00a979"
              />
              <path
                d="M10.35 15.55a.91.91 0 01-.53-.25l-2.48-2.71a.77.77 0 01.13-.9.7.7 0 011 0l1.9 2.11L15 9a.7.7 0 011 0 .68.68 0 01.21.49.67.67 0 01-.21.49l-5 5.37a.77.77 0 01-.65.2zm-2.58-3.21L10.18 15a.54.54 0 00.18.1c.07 0 .1 0 .18-.1l5-5.37a.19.19 0 00.07-.14.18.18 0 00-.06-.13.19.19 0 00-.28 0l-5 5.21L8.1 12a.2.2 0 00-.28 0 .28.28 0 00-.05.34z"
                fill="#00a979"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2"
            >
              Available when you need care
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
